src/Analyser/MutatingScope.php

```diff
@@ -2610,7 +2614,7 @@ private function enterFunctionLike(
 		foreach (ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getParameters() as $parameter) {
 			$parameterType = $parameter->getType();
 			if ($parameter->isVariadic()) {
-				if ($this->phpVersion->supportsNamedArguments()) {
+				if ($this->phpVersion->supportsNamedArguments() && $functionReflection->acceptsNamedArguments()) {
```

Comment:
There is a simillar check in [`getFunctionType()`](https://github.com/phpstan/phpstan-src/blob/cf8832c79b51cc3d8751b1f3651cfcf31ec5298c/src/Analyser/MutatingScope.php#L3051-L3065). I'm not sure whether I need to change anything there, as it seems to work without it.